components/sidebar/channel_filter/channel_filter.tsx

```diff
@@ -2,25 +2,80 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import classNames from 'classnames';
 
 type Props = {
-
+    unreadFilterEnabled: boolean;
+    actions: {
+        setUnreadFilterEnabled: (enabled: boolean) => void;
+    };
 };
 
 type State = {
 
 };
 
 export default class ChannelFilter extends React.PureComponent<Props, State> {
+    toggleUnreadFilter = () => {
+        const {unreadFilterEnabled} = this.props;
+        this.props.actions.setUnreadFilterEnabled(!unreadFilterEnabled);
+    }
+
     render() {
+        const {unreadFilterEnabled} = this.props;
+
+        let filterTitle = (
+            <FormattedMessage
+                id='sidebar_left.channel_filter.viewing'
+                defaultMessage='VIEWING:'
```

Comment:
Would it make sense to capitalize these with a `text-transform` instead of having the translation strings be in all caps? 0/5

Reply:
I'd say no, because then different languages can't choose whether capitalization matters to them or not. This is more customizable per language, and is also much easier to understand, since a `text-transform` is hidden away in the CSS and isn't immediately apparent.

Reply:
I don't think that'd necessarily be something we care to change per-language though since using all caps is an aesthetic decision. 0/5 still though

Reply:
I prefer the text-transform for strings usually FWIW, but probably not a blocker at this point.